de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/modelaccess/PropertyResourceTest.java

```diff
@@ -127,5 +133,34 @@ public void testPropertyComposedPut() throws JAXBException {
 	@Test
 	public void testErrorResponses() {
 		assertNotFoundResponse(getTestRequestBuilder(ModelAccessResource.PROPERTY + "/unknown").get());
+		
+		assertNotFoundResponse(getTestRequestBuilder(ModelAccessResource.PROPERTY + "/unknown/" + PropertyResource.RESOURCE).get());
+		// A boolean property doesn't have a resource
+		assertBadRequestResponse(
+				getTestRequestBuilder(ModelAccessResource.PROPERTY + "/" + beanBool.getUuid().toString() + "/" + PropertyResource.RESOURCE).get(),
+				PropertyResource.PROPERTY_DOES_NOT_CONTAIN_A_SERVEABLE_RESOURCE);
+	}
+	
+	@Test
+	public void testResourceGet() throws CoreException {
+		// Initial the bean does not contain a valid resource
+		assertBadRequestResponse(
+				getTestRequestBuilder(ModelAccessResource.PROPERTY + "/" + beanResource.getUuid().toString() + "/" + PropertyResource.RESOURCE).get(),
+				PropertyResource.PROPERTY_DOES_NOT_CONTAIN_A_SERVEABLE_RESOURCE);
+		
+		// Now create a valid file
+		IFile file = ed.getResourceSet().getProject().getFile("file[2].xls");
+		file.create(new ByteArrayInputStream(TEST_STRING.getBytes()), IResource.FORCE, null);
```

Comment:
`.xls` implies a binary file, but you write a text string into it. Should we maybe test it on a real binary file, and then compare bytes?

Reply:
Thanks for the review, I hope I changed it accordingly